server/aesdsocket.c

```diff
@@ -1,82 +1,203 @@
 */
 
-// Open a stream socket bound to port 9000, fail and exit(-1) on any failure.
-// Listens for and accepts connection
-// close the file descriptor both from the socket and accept to avoid memory leaks.
-// consider shutting down
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <errno.h>
+#include <netinet/in.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <syslog.h>
+#include <net/if.h>
+#include <sys/stat.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <arpa/inet.h>
+#include <signal.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <string.h>
+#include <netdb.h>
 
-#include<sys/types.h>
-#include<sys/socket.h>
-#include<netdb.h>
-#include<stdio.h>
-#include<stdlib.h>
-#include<unistd.h>
-#include<string.h>
-#include<sys/stat.h>
-#include<fcntl.h>
-#include<errno.h>
-#include<signal.h>
-#include<syslog.h>
-#include<arpa/inet.h>
-#include<stdbool.h>
+#define TIMESTAMP_SIZE 50
 
-#define BUFFER_SIZE 1024
-
-int sockfd;
 int fd;
-/* Perform Signal handler on SIGINT and SIGTERM */
-// Obtained guidance from Swapnil Ghonge on how to device this handler function
-void signal_handler(int signo)
+int sockfd;
+pthread_mutex_t mutex_lock;
+// To perform multithreaded applications, 
+typedef struct node
 {
-    printf("%d signal caught", signo);
-    if ((signo == SIGINT) || (signo == SIGTERM)) {
-        close(fd);
-        close(sockfd);
-        remove("/var/tmp/aesdsocketdata");
-        exit(0);
-    }
-}
-
-int main(int argc, char **argv){
+	pthread_t thread;
+	int clientfd;
+	bool thread_complete_status;
+	char client_ipaddress[INET6_ADDRSTRLEN];
+	TAILQ_ENTRY(node) entries;
+}thread_data_t;
 
-char buffer[BUFFER_SIZE] = {0};
-
-signal(SIGINT, signal_handler);
-signal(SIGTERM, signal_handler);
-/******************************************************************************************** Create the Socket **************************************************************************************/
-syslog(LOG_USER, "Socket Creation");
-sockfd = socket(AF_INET,SOCK_STREAM,0);
-if(sockfd == -1){
-   perror("Socket Not created Properly");
-   exit(-1);
+typedef TAILQ_HEAD(head_s, node) head_t;
+head_t head;
+// Assignment instruction 2b - Append timestamp in 24hours format
+// parameters : 	None
+// Returns    :	None
+// References	:	https://geeksforgeeks.org/strftime-function-in-c/
+void append_time_stamp()
+{
+	time_t t;
+	struct tm *tmp;
+	char MY_TIME[TIMESTAMP_SIZE];
+	time(&t);
+	tmp = localtime(&t);
+	strftime(MY_TIME, sizeof(MY_TIME), "timestamp: %Y-%m-%d %H:%M:%S\r\n", tmp);
+	printf("%s", MY_TIME);
+	lseek(fd, 0, SEEK_END);						// Move the current file to the end
+	int write_bytes_count = write(fd, MY_TIME, strlen(MY_TIME));
+	if (write_bytes_count != strlen(MY_TIME)) {
+		printf("write unsuccessful\n");
+	}
```

Comment:
@srki3050 are all of these operations safe to do from a signal handler?  What does it mean to manipulate `fd` here and what will happen if this file descriptor was used outside the handler?
Also check the assignment for requriements around locking.